composer.json

```diff
@@ -0,0 +1,58 @@
+{
+    "name": "grongor/kafka-rest-client",
+    "description": "Client for Confluent REST Proxy of Apache Kafka",
+    "type": "library",
+    "license": "MIT",
+    "keywords": ["kafka", "confluent", "proxy", "rest", "client", "librdkafka", "rdkafka"],
+    "config": {
+        "sort-packages": true
+    },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/grongor/curl-client"
+        }
+    ],
+    "require": {
+        "php": "^7.2|^8.0",
+        "beberlei/assert": "^3.2",
+        "jms/serializer": "^3.3",
+        "lcobucci/clock": "^1.2",
+        "php-http/discovery": "^1.0",
```

Comment:
IMO we can avoid any of these to avoid magic. Any particular reason to include it?

Reply:
Simple ... it's the recommended way mentioned on http://docs.php-http.org/en/latest/ 

And to be honest I don't mind it. You are able to pass your own implementations, this is just a fallback that works just by having the appropriate library installed, which makes sense. I see it like some of the "magic" we use while building Kernel, where everything automagically wires together.

Reply:
You scared me, I was afraid it's recommended there :D I don't see it recomended, it's only one of their components. 

The issue is it allows you to bypass DI by creating magical gateway through which hordes of dependencies can invade unspotted.

If one wants to abuse DI, he should install it into his system and pass into our constructor. We should not encourage not being defensive.  
Also, less dependencies, the better.

Reply:
http://docs.php-http.org/en/latest/httplug/library-developers.html#discovery

Reply:
I suppose they won't accept PR recomending removing their library. It still applies it's malpractice 👆